src/main/java/ch/njol/skript/expressions/ExprHealAmount.java

```diff
@@ -0,0 +1,120 @@
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.expressions;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityRegainHealthEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+
+@Name("Heal Amount")
+@Description("The amount of a healing event, which can be changed.")
+@Examples({"increase heal amount by 2",
+	"remove 0.5 from heal amount"})
+@Since("INSERT VERSION")
+@Events("heal")
+
+public class ExprHealAmount extends SimpleExpression<Number> {
+	
+	static {
+		Skript.registerExpression(ExprHealAmount.class, Number.class, ExpressionType.SIMPLE, "[the] heal amount");
+	}
+	
+	@SuppressWarnings("null")
+	private Kleenean delay;
+	
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(EntityRegainHealthEvent.class)) {
+			Skript.error("The expression 'heal amount' may only be used in heal event", ErrorQuality.SEMANTIC_ERROR);
+			return false;
+		}
+		delay = isDelayed;
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	protected Number[] get(Event e) {
+		return new Number[]{((EntityRegainHealthEvent) e).getAmount()};
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (delay != Kleenean.FALSE) {
+			Skript.error("The heal amount cannot be changed after the event has already passed");
+			return null;
+		}
+		if (mode == Changer.ChangeMode.REMOVE_ALL || mode == Changer.ChangeMode.RESET)
+			return null;
+		return new Class[]{Number.class};
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		double value = delta == null ? 0 : ((Number) delta[0]).doubleValue();
+		switch (mode) {
+			case SET:
+			case DELETE:
```

Comment:
Why would SET be equal to DELETE here?

Reply:
Because if `delta` is null, the value will be set to 0:
`double value = delta == null ? 0 : ((Number) delta[0]).doubleValue();`

Reply:
Because if `delta` is null, the value will be set to 0:
`double value = delta == null ? 0 : ((Number) delta[0]).doubleValue();`